rotary-encoder/rotary_encoder.go

```diff
@@ -0,0 +1,66 @@
+//go:build tinygo
+
+// Implementation based on:
+// https://gist.github.com/aykevl/3fc1683ed77bb0a9c07559dfe857304a
+
+package rotary_encoder
+
+import (
+	"machine"
+)
+
+var (
+	states = []int8{0, -1, 1, 0, 1, 0, 0, -1, -1, 0, 0, 1, 0, 1, -1, 0}
+)
+
+func New(pinA, pinB machine.Pin) *Device {
+	return &Device{pinA: pinA, pinB: pinB, oldAB: 0b00000011}
+}
+
+type Device struct {
+	pinA machine.Pin
+	pinB machine.Pin
+
+	precision int
+
+	oldAB int
+	value int
```

Comment:
`value` needs to be accessed using volatile operations. Please use the runtime/volatile package for this. Without it, you'll run into undefined behavior (this is probably why you needed the `time.Sleep` function).

(Yes, this means the gist and the original example on circuitsathome.com are both subtly incorrect).